src/muya/lib/contentState/tableSelectCellsCtrl.js

```diff
@@ -0,0 +1,248 @@
+import { getAllTableCells, getIndex } from './tableDragBarCtrl'
+
+const tableSelectCellsCtrl = ContentState => {
+  ContentState.prototype.handleCellMouseDown = function (event) {
+    const { eventCenter } = this.muya
+    const { target } = event
+    const cell = target.closest('th') || target.closest('td')
+    const tableId = target.closest('table').id
+    const row = getIndex('left', cell)
+    const column = getIndex('bottom', cell)
+    this.cellSelectInfo = {
+      tableId,
+      anchor: {
+        key: cell.id,
+        row,
+        column
+      },
+      focus: null,
+      isStartSelect: false,
+      cells: getAllTableCells(tableId),
+      selectedCells: []
+    }
+
+    const mouseMoveId = eventCenter.attachDOMEvent(document.body, 'mousemove', this.handleCellMouseMove.bind(this))
+    const mouseUpId = eventCenter.attachDOMEvent(document.body, 'mouseup', this.handleCellMouseUp.bind(this))
+    this.cellSelectEventIds.push(mouseMoveId, mouseUpId)
+  }
+
+  ContentState.prototype.handleCellMouseMove = function (event) {
+    const { target } = event
+    const cell = target.closest('th') || target.closest('td')
+    const table = target.closest('table')
+    const isOverSameTableCell = cell && table && table.id === this.cellSelectInfo.tableId
+    if (isOverSameTableCell && cell.id !== this.cellSelectInfo.anchor.key) {
+      this.cellSelectInfo.isStartSelect = true
+      this.muya.blur()
+    }
+    if (isOverSameTableCell && this.cellSelectInfo.isStartSelect) {
+      const row = getIndex('left', cell)
+      const column = getIndex('bottom', cell)
+      this.cellSelectInfo.focus = {
+        key: cell.key,
+        row,
+        column
+      }
+    } else {
+      this.cellSelectInfo.focus = null
+    }
+
+    this.calculateSelectedCells()
+    this.setSelectedCellsStyle()
+  }
+
+  ContentState.prototype.handleCellMouseUp = function (event) {
+    const { eventCenter } = this.muya
+    for (const id of this.cellSelectEventIds) {
+      eventCenter.detachDOMEvent(id)
+    }
+    this.cellSelectEventIds = []
+    if (this.cellSelectInfo && this.cellSelectInfo.isStartSelect) {
+      event.preventDefault()
+      const { tableId, selectedCells, anchor, focus } = this.cellSelectInfo
+      setTimeout(() => {
```

Comment:
Can you explain why do you use a timeout here?

Reply:
// We need to handle this after click event, because click event is emited after mouseup(mouseup will be followed by a click envent), but we set
      // the `selectedTableCells` to null when click event emited.

Reply:
Added comments.